_posts/2025-10-06-foremanctl-10---deploying-foreman-in-containers-reaches-first-milestone.md

```diff
@@ -0,0 +1,62 @@
+---
+layout: post
+title: foremanctl 1.0 - deploying Foreman in Containers reaches first milestone
+date: 2025-10-06  7:47:50
+author: Archana Kumari
+tags:
+- foreman
+- foremanctl
+- containers
+- quadlet
+- podman
+- systemd
+- release
+- katello
+- pulp
+---
+
+Over the past few months, we’ve been working on a new way to deploy and manage Foreman and its components (like Katello, Candlepin, and Pulp) using **containers**.
+Today, we’re happy to share that **foremanctl 1.0** has been tagged, marking our first milestone toward a fully containerized Foreman experience.
+
+<!--more-->
+
+### What You Can Do with 1.0
+
+With this first release, you can already:
+
+- Set up a basic Foreman + Katello environment using containers.
+- Use an external database configuration.
+- Manage services with systemd and Quadlet, making it easier to start, stop, and maintain the setup.
+
+Under the hood, `foremanctl` generates **Podman Quadlet** units that integrate directly with **systemd**, providing a consistent and declarative way to run Foreman services as containers — without relying on traditional installer RPMs.
+
+If you’d like to try it out, follow the updated [Quickstart Guide for 3.16](https://theforeman.org/manuals/3.16/quickstart_guide.html), which now includes step-by-step instructions for installing and running `foremanctl`. For detailed changes and improvements, see the [1.0.0 release notes](https://github.com/theforeman/foremanctl/releases/tag/1.0.0).
+
+---
+
+### What’s Still in Progress
+
+This release is an important first step, but it’s not yet a full replacement for the traditional installer setup. We’re continuing to work on:
+
+- Containerizing Smart Proxy  
+- Adding support for managing plugins (currently static list)  
+- Improving SELinux and networking integration  
+- Extending system requirement checks and configuration flexibility  
+
+> **Note:** Smart Proxy services (like REX, Ansible, and DNS/DHCP) are not yet available in the release.
+
+You can follow progress and upcoming work items in the [foremanctl milestone 2.0 roadmap](https://github.com/theforeman/foremanctl/milestone/2).
+Upcoming releases will build on this foundation: **3.16** as an *alpha milestone*, **3.17** as *beta*, and **3.18** targeting a *GA-ready containerized deployment flow*.
```

Comment:
Not sure what the convention on promising features in future releases are but I'm wondering if the wording here should be more careful: Rather than state with certainty that these releases will represent these milestones, perhaps use wording like "is expected", "is planned", "should", etc.? Or just add a statement that plans can change?

Reply:
I'd tone it down to a desire. Perhaps use `aim` instead of `targeting`?